internal/sinkcores/outputcore/override.go

```diff
@@ -88,10 +83,23 @@ func (c *overrideCore) Check(ent zapcore.Entry, ce *zapcore.CheckedEntry) *zapco
 			continue
 		}
 
+		// If we found a matching override, use its level check
 		if o.Level.Enabled(ent.Level) {
 			return c.Core.Check(ent, ce)
 		}
+		return ce
+	}
+
+	// If no overrides matched or the override allowed it, check the base level
+	if !c.level.Enabled(ent.Level) {
+		return ce
+	}
+
+	// Finally check if the core itself allows this level
+	if !c.Core.Enabled(ent.Level) {
+		return ce
 	}
 
-	return ce
+	// allowed!
+	return c.Core.Check(ent, ce)
```

Comment:
I'm not sure why you need to change this. It should still work as before, or you need to update the docstrings to match the new invariants.

And even then, I'm not sure why you changed how these invariants work, couldn't you just move the block down?

Reply:
The logic / invariants are the same, it's just phrased in terms of identifying negatives instead of positives. But I've reverted it back to the OG so the diff is smaller.